wallet/trezor-common/src/tests/mod.rs

```diff
@@ -0,0 +1,557 @@
+    fn from(value: chain::tokens::TokenIssuance) -> Self {
+        match value {
+            chain::tokens::TokenIssuance::V1(data) => Self::V1(crate::TokenIssuanceV1 {
+                token_ticker: data.token_ticker,
+                number_of_decimals: data.number_of_decimals,
+                metadata_uri: data.metadata_uri,
+                total_supply: data.total_supply.into(),
+                authority: data.authority.into(),
+                is_freezable: data.is_freezable.into(),
+            }),
+        }
+    }
+}
+
+impl From<chain::TxOutput> for crate::TxOutput {
+    fn from(value: chain::TxOutput) -> Self {
+        match value {
+            chain::TxOutput::Transfer(value, dest) => Self::Transfer(value.into(), dest.into()),
+            chain::TxOutput::LockThenTransfer(value, dest, lock) => {
+                Self::LockThenTransfer(value.into(), dest.into(), lock.into())
+            }
+            chain::TxOutput::Burn(amount) => Self::Burn(amount.into()),
+            chain::TxOutput::DataDeposit(data) => Self::DataDeposit(data),
+            chain::TxOutput::CreateDelegationId(dest, pool_id) => {
+                Self::CreateDelegationId(dest.into(), pool_id.to_hash().into())
+            }
+            chain::TxOutput::DelegateStaking(amount, delegation_id) => {
+                Self::DelegateStaking(amount.into(), delegation_id.to_hash().into())
+            }
+            chain::TxOutput::ProduceBlockFromStake(dest, pool_id) => {
+                Self::ProduceBlockFromStake(dest.into(), pool_id.to_hash().into())
+            }
+            chain::TxOutput::Htlc(value, lock) => Self::Htlc(value.into(), (*lock).into()),
+            chain::TxOutput::CreateOrder(data) => Self::CreateOrder((*data).into()),
+            chain::TxOutput::CreateStakePool(pool_id, data) => {
+                Self::CreateStakePool(pool_id.to_hash().into(), (*data).into())
+            }
+            chain::TxOutput::IssueNft(token_id, data, dest) => {
+                Self::IssueNft(token_id.to_hash().into(), (*data).into(), dest.into())
+            }
+            chain::TxOutput::IssueFungibleToken(data) => Self::IssueFungibleToken((*data).into()),
+        }
+    }
+}
+
+fn make_random_destination(rng: &mut (impl Rng + CryptoRng)) -> chain::Destination {
+    match rng.gen_range(0..=4) {
+        0 => chain::Destination::AnyoneCanSpend,
+        1 => chain::Destination::PublicKey(
+            crypto::key::PrivateKey::new_from_rng(rng, KeyKind::Secp256k1Schnorr).1,
+        ),
+        2 => chain::Destination::ScriptHash(primitives::H256(rng.gen()).into()),
+        3 => chain::Destination::ClassicMultisig(
+            (&crypto::key::PrivateKey::new_from_rng(rng, KeyKind::Secp256k1Schnorr).1).into(),
+        ),
+        _ => chain::Destination::PublicKeyHash(
+            (&crypto::key::PrivateKey::new_from_rng(rng, KeyKind::Secp256k1Schnorr).1).into(),
+        ),
+    }
+}
```

Comment:
I suggest deriving `strum::EnumDiscriminants` for the existing enum types, like I did in [1832](https://github.com/mintlayer/mintlayer-core/pull/1832) for `Destination`:
```
#[derive(..., EnumDiscriminants)]
#[strum_discriminants(name(DestinationTag), derive(EnumIter))]
pub enum Destination {
...
```
This will define an additional enum called `DestinationTag`, which will be like `Destination` but without the data. And this `DestinationTag` will implement `EnumIter`, so it'll be possible to iterate over its values.
Then you'd be able to write:
```
let dest_tag = DestinationTag::iter().choose(rng).unwrap();
match dest_tag {
    DestinationTag::PublicKey => {
        chain::Destination::PublicKey(
            crypto::key::PrivateKey::new_from_rng(rng, KeyKind::Secp256k1Schnorr).1,
        )
    },
    ...
}
```
So, if/when a new variant is added to the enum, we'll get a compilation failure here.